apps/trading/components/market-banner/market-successor-proposal-banner.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState } from 'react';
+import type {
+  SuccessorProposalListFieldsFragment,
+  NewMarketSuccessorFieldsFragment,
+} from '@vegaprotocol/proposals';
+import { useSuccessorProposalsListQuery } from '@vegaprotocol/proposals';
+import {
+  ExternalLink,
+  Intent,
+  NotificationBanner,
+} from '@vegaprotocol/ui-toolkit';
+import { t } from '@vegaprotocol/i18n';
+import { DApp, TOKEN_PROPOSAL, useLinks } from '@vegaprotocol/environment';
+
+export const MarketSuccessorProposalBanner = ({
+  marketId,
+}: {
+  marketId?: string;
+}) => {
+  const { data: proposals } = useSuccessorProposalsListQuery({
+    skip: !marketId,
+  });
+  const successors =
+    proposals?.proposalsConnection?.edges
+      ?.map((item) => item?.node as SuccessorProposalListFieldsFragment)
+      .filter(
+        (item: SuccessorProposalListFieldsFragment) =>
+          (item.terms?.change as NewMarketSuccessorFieldsFragment)
+            ?.successorConfiguration?.parentMarketId === marketId
+      ) ?? [];
+  const [visible, setVisible] = useState<boolean>(true);
```

Comment:
chore (non blocking): Preferably just infer this

Reply:
Cannot easily infer this, since proposal change is union type and possibly can have different shape depending on type of constituent of the union

Reply:
done